lib/db/schema/patch-042-043.sql

```diff
@@ -0,0 +1,20 @@
+CREATE PROCEDURE `prune_2` (IN pruneTokensMaxAge BIGINT UNSIGNED)
+BEGIN
+    -- try and obtain the prune lock
+    SELECT @lockAcquired:=GET_LOCK('fxa-auth-server.prune-lock', 3);
+
+    IF @lockAcquired THEN
+
+      DELETE FROM accountResetTokens WHERE createdAt < pruneTokensMaxAge ORDER BY createdAt LIMIT 10000;
+      DELETE FROM passwordForgotTokens WHERE createdAt < pruneTokensMaxAge ORDER BY createdAt LIMIT 10000;
+      DELETE FROM passwordChangeTokens WHERE createdAt < pruneTokensMaxAge ORDER BY createdAt LIMIT 10000;
+      DELETE FROM unblockCodes WHERE createdAt < pruneTokensMaxAge ORDER BY createdAt LIMIT 10000;
```

Comment:
You'll need to add an index on `createdAt` on the `unblockCodes` table in order for this query to be efficient.  Compare:

```
EXPLAIN DELETE FROM unblockCodes WHERE createdAt < 123456 ORDER BY createdAt LIMIT 10000;
+----+-------------+--------------+------+---------------+------+---------+------+------+-----------------------------+
| id | select_type | table        | type | possible_keys | key  | key_len | ref  | rows | Extra                       |
+----+-------------+--------------+------+---------------+------+---------+------+------+-----------------------------+
|  1 | SIMPLE      | unblockCodes | ALL  | NULL          | NULL | NULL    | NULL |  712 | Using where; Using filesort |
+----+-------------+--------------+------+---------------+------+---------+------+------+-----------------------------+
```

And

```
 EXPLAIN DELETE FROM accountResetTokens WHERE createdAt < 123456 ORDER BY createdAt LIMIT 10000;
+----+-------------+--------------------+-------+---------------+-----------+---------+-------+------+-------------+
| id | select_type | table              | type  | possible_keys | key       | key_len | ref   | rows | Extra       |
+----+-------------+--------------------+-------+---------------+-----------+---------+-------+------+-------------+
|  1 | SIMPLE      | accountResetTokens | range | createdAt     | createdAt | 8       | const |    1 | Using where |
+----+-------------+--------------------+-------+---------------+-----------+---------+-------+------+-------------+
```

Something like this should do the trick, and can be done without downtime:

```
CREATE INDEX unblockCodes_createdAt
ON unblockCodes (createdAt)
ALGORITHM = INPLACE LOCK = NONE;
```